### PR TITLE
Adding Tests

### PR DESCRIPTION
**Pull request by adamcharnock**
I thought I'd spend a few hours writing some tests for django-su today.A total of 23 tests giving 86% code coverage. See:

https://travis-ci.org/adamcharnock/django-su/branches
https://coveralls.io/builds/2968404

The main result of this is a refactoring of the backend in ``backends.py`` to make it a little more robust.

@bashu: Do you have time available to review and potentially merge this?

### CODE REVIEW

**Comment by bashu** [comment]
@adamcharnock yes I am looking...

File: django_su/backends.py [hunk 1]
```diff
@@ -13,7 +13,21 @@ class SuBackend(object):
     def authenticate(self, su=False, pk=None, **credentials):
         if not su:
             return None
-        return User.objects.get(pk=pk)
+
+        try:
+            User._meta.pk.get_prep_value(pk)
+        except ValueError:
+            return None
```
Comment (adamcharnock) [inline django_su/backends.py]: I'd be willing to accept that the above validation check doesn't need to be done, or could perhaps be done better. This seemed to be the best way I could find though.

**Comment by bashu** [comment]
@adamcharnock offtopic but can you enable this coveralls.io for tz_detect too?

File: django_su/backends.py (hunk 1 above)
Comment (bashu) [inline django_su/backends.py]: can be done as 
```python
        try:
            user = User._default_manager.get(pk=user_id)
        except (User.DoesNotExist, ValueError):
            return None
```

**Comment by adamcharnock** [comment]
Thanks @bashu! Also, just enabled coveralls for tz_detect:
https://coveralls.io/r/adamcharnock/django-tz-detect

I'll update travis.yml shortly too.

**Pull request merged**